File: vol1/ch6/LeetCode965.py
class Solution:
    def isUnivalTree(self, root: Optional[TreeNode]) -> bool:
        if root==None:return True
        else: return self.preorder(root,root.val)

    def preorder(self,root,d):
        if root==None:
            return True
        else:
            if root.val!=d:return False
            leftans=self.preorder(root.left,d)
            if not leftans: return False
            else: return self.preorder(root.right,d)

File: vol1/ch6/LeetCode965.cpp
class Solution {
public:
    bool isUnivalTree(TreeNode* root) {
        if(root==NULL) return true;
        else return preorder(root,root->val);
    }
    bool preorder(TreeNode*root,int d) {
        if(root==NULL)
            return true;
        else {
            if(root->val!=d) return false;
            bool leftans=preorder(root->left,d);
            if(!leftans) return false;
            else return preorder(root->right,d);
        }
    }
};

